catkin_tools/execution/job_server.py

```diff
@@ -162,9 +161,8 @@ def _set_max_jobs(cls, max_jobs):
         # Read all possible tokens from the pipe
         try:
             os.read(cls._job_pipe[0], cls._max_jobs)
-        except OSError as e:
-            if e.errno != errno.EINTR:
-                raise
+        except (BlockingIOError, InterruptedError):
+            pass
```

Comment:
Both of these derive from OSError. Any other OSError exceptions will raise through, as the behaviour was before.